### PR TITLE
[CLOUD] Fix upload images

### PR DESCRIPTION
**Pull request by cfconrad**
1) Detection of already uploaded images where where broken for azure.
   We search now for the corresponding resource group and retrieves
   the first image.

2) Increase upload timeout to 2h for azure.

3) Install aws-cli from CloudModule instead of CloudTools repo, duo to
   incompatibilities between aws from pip and python-ec2uploadimg from
   CloudModule.

4) Fix upload_img.pm and set PUBLIC_CLOUD_IMAGE_ID with the retrieved
   ID from $provider->find_img().

- Verification run: 
   Azure: http://cfconrad-vm.qa.suse.de/tests/1579 (existing image)
   Azure: http://cfconrad-vm.qa.suse.de/tests/1578
   EC2: http://cfconrad-vm.qa.suse.de/tests/1576 (existing image)
   EC2: http://cfconrad-vm.qa.suse.de/tests/1574

### CODE REVIEW

**Comment by asmorodskyi** [comment]
commit message contains "uploaded images where where broken"

File: lib/publiccloud/azure.pm [hunk 1]
```diff
@@ -38,6 +48,21 @@ sub find_img {
...
 }
 
+sub find_img {
+    my ($self, $name) = @_;
+
+    $name = $self->filename_to_resgroup($name);
+    return unless $self->find_resgroup($name);
+
+    my $images = decode_json(script_output("az image list --resource-group '$name'"));
+    for my $i (@{$images}) {
```
Comment (asmorodskyi) [inline lib/publiccloud/azure.pm]: didn't get why you need a loop here ? 
`return $images[0]->{name}` would do the same as code in lines 58-60 .
Or you overlook if statement inside a loop ?

Comment (cfconrad) [reply lib/publiccloud/azure.pm]: No it's actually a left over, the --resource-group switch was added later.
Thanks for pointing that out!

**Comment by cfconrad** [comment]
Prove run: http://cfconrad-vm.qa.suse.de/tests/1581

**Pull request merged**